File: py/Monthly calculator/summation.py
fileData = ''
fileName = input('File name: ') or 'data'
with open(fileName + '.txt', 'r') as myfile:
    fileData = ''.join(myfile.readlines())

fileDataArray = fileData.split('\n')
monthlyTotal = []
totalsum = 0
yeartotal = 0
month = int(fileData.split('\n')[0].split(':')[0].split('/')[1])
year = int(fileData.split('\n')[0].split(':')[0].split('/')[2])
for i in fileDataArray:
    summ = 0
    if(i.count('/') < 2):
        yeartotal = yeartotal + totalsum
        totalsum = 0
        monthlyTotal.append(i)
        continue
    if(i.count(':') > 1):
        summ = float(i.split(' : ')[2].replace(',', ''))
        totalsum = totalsum + summ
        monthlyTotal.append(i)
        continue
    # print(i)
    loc = i.split(' : ')[1]
    if(len(loc.split(' + ')) > 1 and loc.split(' + ')[0] != ''):
        for j in loc.split(' + '):
            summ = summ + float(j.replace(',', ''))
        formated = '{:,.2f}'.format(summ).replace('.00', '')
        monthlyTotal.append(i + ' : ' + formated)
        totalsum = totalsum + summ
    elif(loc.split(' + ')[0] != ''):
        monthlyTotal.append(i + ' : ' + loc.split(' + ')[0])
        totalsum = totalsum + float(loc.split(' + ')[0].replace(',', ''))
    else:
        monthlyTotal.append(i)
yeartotal = '{:,.2f}'.format(yeartotal + totalsum).replace('.00', '')
formated = '{:,.2f}'.format(totalsum).replace('.00', '')
monthlyTotal.append('\n------------------------ ' + formated +
                    ' : ' + formated + ' -------------------------\n')

newMonth = []
end = 31
year = year + 1 if month == 12 else year
month = month + 1 if month < 12 else 1

if(month in [1, 3, 5, 7, 8, 10, 12]):
    end = 32
elif(month == 2):
    end = 29
for i in range(1, end):
    newMonth.append(str(i) + '/' + str(month) + '/' + str(year) + ' : ')
newMonth.append('\n------------------------  :  -------------------------')
print('\n'.join(monthlyTotal))
print('\n'.join(newMonth))
print(yeartotal)
f = open('m.total.txt', 'w')
f.write('\n'.join(monthlyTotal))
f.close()
f = open('nm.txt', 'w')
f.write('\n'.join(newMonth))
f.close()

File: js/Monthly Calculator/summation.js
const fse = require('fs-extra')

let fileName = (process.argv[2]) ? process.argv[2] : 'data'
let fileData = fse.readFileSync(fileName + '.txt', 'utf8')


let fileDataArray = fileData.split('\r\n')
let monthlyTotal = []
let totalsum = 0
let yeartotal = 0
let month = parseInt(fileData.split('\n')[0].split(':')[0].split('/')[1])
let year = parseInt(fileData.split('\n')[0].split(':')[0].split('/')[2])

for (let i of fileDataArray) {
    let sum = 0
    if (i.match('/') < 2) {
        yeartotal = yeartotal + totalsum
        totalsum = 0
        monthlyTotal.push(i)
        continue
    }
    if (i.match(':') > 1) {
        sum = parseFloat(i.split(' : ')[2].replace(',', ''))
        totalsum = totalsum + sum
        monthlyTotal.push(i)
        continue
    }
    // console.log(i)
    loc = i.split(' : ')[1]
    if ((loc.split(' + ')).length > 1 && loc.split(' + ')[0] != '') {
        for (let j of loc.split(' + '))
            sum = sum + parseFloat(j.replace(',', ''))
        formated = (sum + '').replace('.00', '')
        monthlyTotal.push(i + ' : ' + formated)
        totalsum = totalsum + sum
    } else if (loc.split(' + ')[0] != '') {
        monthlyTotal.push(i + ' : ' + loc.split(' + ')[0])
        totalsum = totalsum + parseFloat(loc.split(' + ')[0].replace(',', ''))
    } else
        monthlyTotal.push(i)

    yeartotal = (yeartotal + totalsum + '').replace('.00', '')
}

formated = (totalsum + '').replace('.00', '')
monthlyTotal.push('\n------------------------ ' + formated + ' : ' + formated + ' -------------------------\n')

newMonth = []
end = 31
year = (month == 12) ? year + 1 : year
month = (month < 12) ? month + 1 : 1

if ([1, 3, 5, 7, 8, 10, 12].includes(month))
    end = 32
else if (month == 2)
    end = 29
for (let i = 1; i < end; i++)
    newMonth.push(i + '/' + month + '/' + year + ' : ')
    
newMonth.push('\n------------------------  :  -------------------------')

console.log(totalsum)
fse.writeFileSync('m.total.txt', monthlyTotal.join('\n'))
fse.writeFileSync('nm.txt', newMonth.join('\n'))
